src/api_server/src/token/mod.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2023 by Alibaba.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::*;
+use serde::Deserialize;
+use serde_json::Value;
+use std::sync::Arc;
+use strum_macros::EnumString;
+use tokio::sync::RwLock;
+
+mod simple;
+
+pub trait TokenBroker {
```

Comment:
The use of a trait object is to enable us to implement more specialized token distribution modules, which can support other token formats (such as CWT), use external keys, and possibly implement token signing and distribution in TEE (such as token distribution modules in Intel SGX Enclave).

Reply:
Since the attestation service has a well defined attestation results formats: https://github.com/confidential-containers/attestation-service#attestation-result-format, I think it's safe to say that it's the KBS responsibility to extract claims from it and format them in different formats. So I agree with you, and a KBS deployment can decide which `AttestationTokenBroker` implementation it wants to use to format and distribute the attestation results.